app/models/claims/training_allowance.rb

```diff
@@ -0,0 +1,54 @@
+class Claims::TrainingAllowance
+  def initialize(mentor:, provider:, academic_year:, claim_to_exclude: nil)
+    @mentor = mentor
+    @provider = provider
+    @academic_year = academic_year
+    @claim_to_exclude = claim_to_exclude
+  end
+
+  def training_type
+    @training_type ||= has_initial_training? ? REFRESHER_TRAINING_TYPE : INITIAL_TRAINING_TYPE
+  end
+
+  def total_hours
+    training_type == INITIAL_TRAINING_TYPE ? INITIAL_TRAINING_HOURS : REFRESHER_TRAINING_HOURS
+  end
+
+  def remaining_hours
+    @remaining_hours ||= total_hours - hours_completed
+  end
+
+  def available?
+    remaining_hours.positive?
+  end
+
+  private
+
+  attr_reader :mentor, :provider, :academic_year, :claim_to_exclude
+
+  INITIAL_TRAINING_TYPE = :initial
+  INITIAL_TRAINING_HOURS = 20
+  REFRESHER_TRAINING_TYPE = :refresher
+  REFRESHER_TRAINING_HOURS = 6
+
+  private_constant :INITIAL_TRAINING_TYPE, :INITIAL_TRAINING_HOURS, :REFRESHER_TRAINING_TYPE,
+                   :REFRESHER_TRAINING_HOURS
+
+  def mentor_training_scope
+    @mentor_training_scope ||= provider.mentor_trainings
+                                       .where(mentor:)
+                                       .joins(claim: { claim_window: :academic_year })
+                                       .merge(Claims::Claim.not_draft_status)
+  end
+
+  def has_initial_training?
+    mentor_training_scope.where(academic_years: { ends_on: ..academic_year.starts_on }).exists?
+  end
```

Comment:
Only look at training on claims **which were submitted**. This means ignore `internal_draft` and `draft` statuses.

```suggestion
  def has_initial_training?
    mentor_training_scope
      .where(academic_years: { ends_on: ..academic_year.starts_on })
      .merge(Claims::Claim.not_draft_status)
      .exists?
  end
```

### Maybe tweak the name?

Also, _maybe_ a more descriptive / less ambiguous name for the method would be `has_trained_in_a_previous_year?`. That would avoid any ambiguity because "initial training" would usually mean `mentor_training.where(training_type: :initial)` (which in this case it's not).